src/main/java/Main.java

```diff
@@ -0,0 +1,57 @@
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import com.google.common.io.Files;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+        System.out.println("Please provide file name...");
+        BufferedReader reader =
+                new BufferedReader(new InputStreamReader(System.in));
+        String fileName = reader.readLine();
+        String[] result = findConcatenatedWords(fileName);
+        System.out.println("Longest concatenated word is: " + result[0]);
+        System.out.println("Second longest concatenated word is: " + result[1]);
+        System.out.println("Concatenated words array size is: " + result[2]);
+    }
+
+    private static String[] findConcatenatedWords(String fileName) throws IOException {
```

Comment:
@joej0e can we decompose this method?
please write some unit tests and refactor the solution at the same time